In [1]:
day_of_the_week = input ('enter_day_of_the_week')

if day_of_the_week.lower() == "monday":
    print('it is monday')
else:
    print('it is not monday')

enter_day_of_the_weekmonday
it is monday


b. prompt the user for a day of the week, print out whether the day is a weekday or a weekend

In [2]:
day_of_the_week = input ('enter_day_of_the_week')

if day_of_the_week.lower() in ['saturday', 'sunday']:
    print("It's the weekend")
else:
    print("It's a weekday")

enter_day_of_the_weekmonday
It's a weekday


### c. create variables and make up values for:
the number of hours worked in one week:

In [3]:
hours_worked = 40
hourly_rate = 60
total_paycheck = hourly_rate * hours_worked

if hours_worked > 40:
    ot_rate = hourly_rate * 1.5
    ot_hours = hours_worked - 40
    print((40*hourly_rate) + (ot_hours * ot_rate))
else:
    print(total_paycheck)


2400


Create an integer variable i with a value of 5.

In [4]:
i = 5


Create a while loop that runs so long as i is less than or equal to 15

In [5]:
while i <= 15:
    print(i)
    i += 1

5
6
7
8
9
10
11
12
13
14
15


### Create a while loop that will count by 2's starting with 0 and ending at 100. Follow each number with a new line.

In [33]:
for n in range(102):
    if n % 2 != 0:
        continue
    print(n)
    if n > 102:
        break

0
2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74
76
78
80
82
84
86
88
90
92
94
96
98
100


### Alter your loop to count backwards by 5's from 100 to -10.

In [10]:
for n in reversed(range(-15,105,5)):
    print(n)
    if n < -10:
        break

100
95
90
85
80
75
70
65
60
55
50
45
40
35
30
25
20
15
10
5
0
-5
-10
-15


### Create a while loop that starts at 2, and displays the number squared on each line while the number is less than 1,000,000. Output should equal:
     2
     4
     16
     256
     65536

In [12]:
i = 2
while i < 65537:
    print(i)
    i = i** 2

2
4
16
256
65536


### Write a loop that uses print to create the output shown below.

100
95
90
85
80
75
70
65
60
55
50
45
40
35
30
25
20
15
10
5

In [18]:
for n in reversed(range(0, 105 ,5)):
    print(n)
    if n < 10:
        break

100
95
90
85
80
75
70
65
60
55
50
45
40
35
30
25
20
15
10
5


### For Loops

#### Write some code that prompts the user for a number, then shows a multiplication table up through 10 for that number.

In [20]:
number = input("enter a number:")

for i in range (1,11):
    print(number, 'x', i, '=', number*i)

enter a number:5
5 x 1 = 5
5 x 2 = 55
5 x 3 = 555
5 x 4 = 5555
5 x 5 = 55555
5 x 6 = 555555
5 x 7 = 5555555
5 x 8 = 55555555
5 x 9 = 555555555
5 x 10 = 5555555555


#### Create a for loop that uses print to create the output shown below.

1
22
333
4444
55555
666666
7777777
88888888
999999999


In [24]:
for i in range (1,10):
    print(str(i)*i)

1
22
333
4444
55555
666666
7777777
88888888
999999999


### Break and Continue

#### Write a program that prompts the user for a positive integer. Next write a loop that prints out the numbers from the number the user entered down to 1.

In [43]:
positive_int = int(input('Enter a positive integer:'))

while positive_int > 0:
    print(positive_int)
    positive_int -= 1
    if positive_int < 1:
        break

Enter a positive integer:5
5
4
3
2
1


#### Prompt the user for an odd number between 1 and 50. Use a loop and a break statement to continue prompting the user if they enter invalid input. (Hint: use the isdigit method on strings to determine this). Use a loop and the continue statement to output all the odd numbers between 1 and 50, except for the number the user entered.

Your output should look like this:

Number to skip is: 27

Here is an odd number: 1
Here is an odd number: 3
Here is an odd number: 5
Here is an odd number: 7
Here is an odd number: 9
Here is an odd number: 11
Here is an odd number: 13
Here is an odd number: 15
Here is an odd number: 17
Here is an odd number: 19
Here is an odd number: 21
Here is an odd number: 23
Here is an odd number: 25
Yikes! Skipping number: 27
Here is an odd number: 29
Here is an odd number: 31
Here is an odd number: 33
Here is an odd number: 35
Here is an odd number: 37
Here is an odd number: 39
Here is an odd number: 41
Here is an odd number: 43
Here is an odd number: 45
Here is an odd number: 47
Here is an odd number: 49


In [56]:
wrong_input = True
number = 0
while wrong_input:
    user_input = input('Enter an odd number between: ')
    if user_input.isdigit():
        number = int(user_input)
        if number % 2 == 1 or number <= 0:
            wrong_input = False
        else:
            wrong_input = True

for i in range(1, 50, 2):
    if i == number:
        print(f'Yikes! Skipping number: {i}')
        continue
    print(f'Here is an odd number: {i}')

Enter an odd number between: 25
Here is an odd number: 1
Here is an odd number: 3
Here is an odd number: 5
Here is an odd number: 7
Here is an odd number: 9
Here is an odd number: 11
Here is an odd number: 13
Here is an odd number: 15
Here is an odd number: 17
Here is an odd number: 19
Here is an odd number: 21
Here is an odd number: 23
Yikes! Skipping number: 25
Here is an odd number: 27
Here is an odd number: 29
Here is an odd number: 31
Here is an odd number: 33
Here is an odd number: 35
Here is an odd number: 37
Here is an odd number: 39
Here is an odd number: 41
Here is an odd number: 43
Here is an odd number: 45
Here is an odd number: 47
Here is an odd number: 49


#### Fizzbuzz

One of the most common interview questions for entry-level programmers is the FizzBuzz test. Developed by Imran Ghory, the test is designed to test basic looping and conditional logic skills.

#### Write a program that prints the numbers from 1 to 100.

In [58]:
for number in range(0, 101):
    print(number)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100


#### For multiples of three print "Fizz" instead of the number

In [61]:
for number in range(1,101):
    if number % 3 == 0:
        print('Fizz')
    else:
        print(number)

Fizz
1
2
Fizz
4
5
Fizz
7
8
Fizz
10
11
Fizz
13
14
Fizz
16
17
Fizz
19
20
Fizz
22
23
Fizz
25
26
Fizz
28
29
Fizz
31
32
Fizz
34
35
Fizz
37
38
Fizz
40
41
Fizz
43
44
Fizz
46
47
Fizz
49
50
Fizz
52
53
Fizz
55
56
Fizz
58
59
Fizz
61
62
Fizz
64
65
Fizz
67
68
Fizz
70
71
Fizz
73
74
Fizz
76
77
Fizz
79
80
Fizz
82
83
Fizz
85
86
Fizz
88
89
Fizz
91
92
Fizz
94
95
Fizz
97
98
Fizz
100


#### For the multiples of five print "Buzz".

#### For numbers which are multiples of both three and five print "FizzBuzz".

In [90]:
i = 0
for number in range(1,101):
    print(i)
    i += 1
    if i%3 == 0 and i%5 == 0:
        print('FizzBuzz')
        i += 1
        continue
    elif i %3 == 0:
        print('Buzz')
        i += 1
        continue
    elif i %5 == 0:
        print('Fizz')
        i += 1

0
1
2
Buzz
4
Fizz
6
7
8
Buzz
10
11
Buzz
13
14
FizzBuzz
16
17
Buzz
19
Fizz
21
22
23
Buzz
25
26
Buzz
28
29
FizzBuzz
31
32
Buzz
34
Fizz
36
37
38
Buzz
40
41
Buzz
43
44
FizzBuzz
46
47
Buzz
49
Fizz
51
52
53
Buzz
55
56
Buzz
58
59
FizzBuzz
61
62
Buzz
64
Fizz
66
67
68
Buzz
70
71
Buzz
73
74
FizzBuzz
76
77
Buzz
79
Fizz
81
82
83
Buzz
85
86
Buzz
88
89
FizzBuzz
91
92
Buzz
94
Fizz
96
97
98
Buzz
100
101
Buzz
103
104
FizzBuzz
106
107
Buzz
109
Fizz
111
112
113
Buzz
115
116
Buzz
118
119
FizzBuzz
121
122
Buzz
124
Fizz
126
127
128
Buzz
130
131
Buzz
133
134
FizzBuzz
136
137
Buzz
139
Fizz
141
142
143
Buzz
145
146
Buzz
148


### 4. Display a table of powers.

#### Prompt the user to enter an integer.

In [66]:
i = int(input('Enter an integer:'))

Enter an integer:5


#### Display a table of squares and cubes from 1 to the value entered.

In [95]:
i = int(input('Enter an integer: '))
for n in range(0, i + 1):
    print(f"{n:3}{n**2:5}{n**3:6}")

Enter an integer: 5
  0    0     0
  1    1     1
  2    4     8
  3    9    27
  4   16    64
  5   25   125


#### Ask if the user wants to continue.

In [100]:
i = int(input('Enter an integer: '))
for n in range(0, i + 1):
    print(f"{n:3}{n**2:5}{n**3:6}")
keep_going = input("Do you want to continue?")
    if keep_going = 'yes' or 'Yes' or 'YES':
        i = int(input('Enter an integer:'))
        for n in range(0, i + 1):
            print(f"{n:3}{n**2:5}{n**3:6}")
        break
        

Enter an integer: 5
  0    0     0
  1    1     1
  2    4     8
  3    9    27
  4   16    64
  5   25   125
Do you want to continue?yes


#### Convert given number grades into letter grades

- Prompt the user for a numerical grade from 0 to 100.
- Display the corresponding letter grade.
- Prompt the user to continue.
- Assume that the user will enter valid integers for the grades.
- The application should only continue if the user agrees to.


In [77]:
num_grade = int(input('What is your numerical grade?'))




SyntaxError: invalid syntax (2276696339.py, line 3)